ansible/modules/hashivault/hashivault_secret_engine.py

```diff
@@ -104,7 +104,7 @@ def main():
     argspec['options'] = dict(required=False, type='dict', default={})
     argspec['cas_required'] = dict(required=False, type='bool')
     argspec['max_versions'] = dict(required=False, type='int')
-    module = hashivault_init(argspec)
+    module = hashivault_init(argspec, supports_check_mode=True)
```

Comment:
Hi,
check mode is indeed a dry run. When combined with --diff it will show you what would be changed when run without check mode.
The create/update/delete if and elif statements all have a requirement on check_mode to be false. It seemed to me that that would be enough to prevent anything from being changed or am i missing something?

Reply:
So they do. Odd that this wasn't set before

Reply:
Test failure message is confusing, but it is pointing right to this line. Maybe run `tox -e py39` and see what you get? Running it manually, you may get better output